src/tree/containerizedFunctionApp/ResolvedContainerizedFunctionAppResourceBase.ts

```diff
@@ -0,0 +1,150 @@
+/*---------------------------------------------------------------------------------------------
+*  Copyright (c) Microsoft Corporation. All rights reserved.
+*  Licensed under the MIT License. See License.md in the project root for license information.
+*--------------------------------------------------------------------------------------------*/
+
+import { type Site, type StringDictionary } from "@azure/arm-appservice";
+import { createWebSiteClient } from "@microsoft/vscode-azext-azureappservice";
+import { AppSettingsTreeItem } from "@microsoft/vscode-azext-azureappsettings";
+import { nonNullProp, nonNullValueAndProp, type AzExtTreeItem, type IActionContext, type ISubscriptionContext, type TreeItemIconPath } from "@microsoft/vscode-azext-utils";
+import { type ResolvedAppResourceBase } from "@microsoft/vscode-azext-utils/hostapi";
+import { type ViewPropertiesModel } from "@microsoft/vscode-azureresources-api";
+import { latestGAVersion, tryParseFuncVersion, type FuncVersion } from "../../FuncVersion";
+import { runFromPackageKey } from "../../constants";
+import { ext } from "../../extensionVariables";
+import { parseHostJson, type IParsedHostJson } from "../../funcConfig/host";
+import { envUtils } from "../../utils/envUtils";
+import { treeUtils } from "../../utils/treeUtils";
+import { type ApplicationSettings, type FuncHostRequest } from "../IProjectTreeItem";
+import { ProjectSource } from "../projectContextValues";
+import { ContainerAppSettingsClientProvider } from "./AppSettingsClient";
+import { type ContainerTreeItem } from "./ContainerTreeItem";
+import { FunctionsTreeItem } from "./FunctionsTreeItem";
+import { ImageTreeItem } from "./ImageTreeItem";
+
+export class ResolvedContainerizedFunctionAppResource implements ResolvedAppResourceBase {
+    public site: Site;
+    public maskedValuesToAdd: string[] = [];
+
+    public appSettingsTreeItem: AppSettingsTreeItem;
+    private _functionsTreeItem: FunctionsTreeItem;
+    private _imageTreeItem: ImageTreeItem;
+
+    private _cachedVersion: FuncVersion | undefined;
+    private _cachedHostJson: IParsedHostJson | undefined;
+
+    public readonly source: ProjectSource = ProjectSource.Remote;
+
+    public constructor(site: Site) {
+        this.site = site;
```

Comment:
There's a bunch of things that we mask when we create the `ResolvedFunctionAppResource`. I believe that you should be able to add most of thoss properties here as well, but you'll have to get it from the `Site` rather than `parsedSite`.